code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -158,6 +158,8 @@ MATCHBOXES ARE ALSO IN FANCY.DM
 	var/brightness_on = 1 //Barely enough to see where you're standing, it's a boring old cigarette
 	var/smoketime = 300
 	var/chem_volume = 15
+	var/inside_item = 0 //For whether the cigarette is contained inside another item.
```

Comment:
The reason I went with a var was so that other objects with similar "not a cigarette but functions exactly like one" items could use the same system, if any more were ever made.